controller_manager:
  ros__parameters:
    update_rate: 1000  # Hz

    impedance_controller:
      type: force_control/Impedance_Controller

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
    
    franka_robot_state_broadcaster:
      type: franka_robot_state_broadcaster/FrankaRobotStateBroadcaster

impedance_controller:
  ros__parameters:
    arm_id: panda
    k_gains:
      - 24.0
      - 24.0
      - 24.0
      - 24.0
      - 10.0
      - 6.0
      - 2.0
    # d_gains:
    #   - 2.0
    #   - 2.0
    #   - 2.0
    #   - 1.0
    #   - 1.0
    #   - 1.0
    #   - 0.5
    d_gains:
      - 16.0
      - 16.0
      - 16.0
      - 8.0
      - 8.0
      - 8.0
      - 4.0

franka_robot_state_broadcaster:
  ros__parameters:
    arm_id: panda
